packages/edit-widgets/src/store/selectors.js

```diff
@@ -80,6 +80,27 @@ export const getWidgetAreaForWidgetId = createRegistrySelector(
 	}
 );
 
+/**
+ * Given a child client id, returns the parent widget area block.
+ *
+ * @param {string} clientId The client id of a block in a widget area.
+ *
+ * @return {WPBlock} The widget area block.
+ */
+export const getParentWidgetAreaBlock = createRegistrySelector(
+	( select ) => ( state, clientId ) => {
+		const { getBlock, getBlockName, getBlockParents } = select(
+			blockEditorStore
+		);
+		const blockParents = getBlockParents( clientId );
+		const widgetAreaClientId = blockParents.find(
```

Comment:
The `blockParents` are ordered from top to bottom, and `find` won't traverse the whole array, so I think in essence it'll be pretty much the same. It should in practice always just pick the first array element, but would still work if widget areas had a parent block (not that this will ever happen 😄 ).